### DISCUSSION TITLE
Busy pods got terminated in mid-work in Kubernetes With KEDA v2

### DISCUSSION THREAD

**Discussion by jitendrabehera2** [kedacore/keda#2899] [Q&A / Need Help]
Hello Team,
Recently I have done scaling based on an Azure Service Bus Queue then we have identified a serious issue: busy pods got terminated during scale-down. As the number of messages in the queue decreases, the HPA automatically shuts down some of the pods. Since the HPA does not know which pods are currently doing actual work, it terminated pods in mid-work and forced them to restart. Please do help on this.

**Excepted :**
**It Should wait till the process complete, once process completed then it should terminated.**

**Comment by zroubalik** [comment]
Hi, this is not something that KEDA could do -> it purely depends on Kuberentes. 

You should handle this in your deployment, check for example this post: https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace
